docs/how-tos/create-a-new-event.rst

```diff
@@ -0,0 +1,323 @@
+- Try reusing existing data classes if possible to avoid duplicating data classes. This will help maintain consistency and reduce the chances of introducing errors.
+- Each field in the payload should be documented with a description of what the field represents and the data type it should contain. This will help consumers understand the payload and react to the event. You should be able to justify why each field is included in the payload and how it relates to the event.
+- Use defaults for optional fields in the payload to ensure its consistency in all cases.
+
+Event Definition
+****************
+
+The :term:`Event Definition` should be implemented in the corresponding subdomain module in the ``signals.py`` file. In our example, the event definition for the enrollment event could be:
+
+.. code-block:: python
+
+    # Location openedx_events/learning/signals.py
+    # .. event_type: org.openedx.learning.course.enrollment.created.v1
+    # .. event_name: COURSE_ENROLLMENT_CREATED
+    # .. event_description: emitted when the user's enrollment process is completed.
+    # .. event_data: CourseEnrollmentData
+    COURSE_ENROLLMENT_CREATED = OpenEdxPublicSignal(
+        event_type="org.openedx.learning.course.enrollment.created.v1",
+        data={
+            "enrollment": CourseEnrollmentData,
+        }
+    )
+
+- The event definition should be documented using in-line documentation with at lest ``event_type``, ``event_name``, ``event_description`` and ``event_data``. This will help consumers understand the event and react to it. See :doc:`../reference/in-line-code-annotations-for-an-event` for more information.
+- The :term:`Event Type` should be unique and follow the naming convention for event types specified in the :doc:`../decisions/0002-events-naming-and-versioning` ADR. This is used by consumers to identify the event.
+- The ``event_name`` should be a constant that is used to identify the event in the code.
+- The ``event_description`` should describe what the event is about and why it is triggered.
+- The ``event_data`` should be the payload class that is used to define the data that is included in the event.
+- The ``data`` dictionary should contain the payload class that is used to define the data that is included in the event. This will help consumers understand the event and react to it. Try using a descriptive name for the data field, but keep consistency with the payload class name. Avoid using suffixes like ``_data`` or ``_payload`` in the data field name.
+- The event should be an instance of the ``OpenEdxPublicSignal`` class to ensure that the event is consistent with the Open edX event framework.
+- Receivers should be able to access the event payload in their receivers to react to the event.
+
+.. TODO: add reference to how to add event bus support to the event's payload
```

Comment:
do we have any link already that we could write in? even if it is a PR where this is being written it would point readers in the right direction. If nothing is available, would you consider removing this comment?

Reply:
Here's the PR for the how-to add event bus support for an event: https://github.com/openedx/openedx-events/pull/428. I'm hoping it gets merged soon so we can reference it here before merging.